### PR TITLE
[TabView] Fix bug where removing tabs from tabcollection would not resize tabs

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

There was a condition that only removing the last tab would resize the tabs which was not the right behavior since tabs should always be resized when one gets removed from tab collection.
## Motivation and Context

Closes https://github.com/microsoft/terminal/issues/9822
## How Has This Been Tested?

Added new interaction tests
## Screenshots (if appropriate):

@zadjii-msft @beervoley FYI

### CODE REVIEW

**Comment by StephenLPeters** [comment]
@chingucoding I think these test failures might be expected given your change. Can you take a look?

File: dev/TabView/TabView.cpp [hunk 1]
```diff
@@ -678,12 +686,9 @@ void TabView::OnItemsChanged(winrt::IInspectable const& item)
                 }
 
             }
-            // Last item removed, update sizes
-            // The index of the last element is "Size() - 1", but in TabItems, it is already removed.
             if (TabWidthMode() == winrt::TabViewWidthMode::Equal)
             {
-                m_updateTabWidthOnPointerLeave = true;
-                if (args.Index() == TabItems().Size())
+                if (!m_pointerInTabstrip)
```
Comment (StephenLPeters) [inline dev/TabView/TabView.cpp:691]: I think we also want to resize the items if it was the last item that was removed, even if the pointer is in the tab strip.

**Review by StephenLPeters — APPROVED** [review]

File: dev/TabView/TabView.cpp (hunk 1 above)
Comment (chingucoding) [reply dev/TabView/TabView.cpp]: Good point, that is also the reason of the test failures (all TabView interaction tests ran fine on my machine).

**Comment by beervoley** [comment]
thanks Marcel!!!!!!

**Pull request merged**